server/services/yamlService.js

```diff
@@ -1,13 +1,22 @@
+/* eslint-disable prefer-destructuring */
 const _ = require('lodash');
 const persistentStore = require('../store/persistentStore');
 
-const quayCatalogSourceImage = 'quay.io/operatorframework/upstream-community-operators:latest';
-
 const generateInstallYaml = (serverRequest, serverResponse) => {
   try {
+    let operatorChannel;
+    let operatorName;
+
     const fields = serverRequest.url.split('/');
-    const operatorChannel = fields[2];
-    const operatorName = fields[3].replace('.yaml', '');
+    if (fields.length === 4) {
```

Comment:
Is there a possible situation where you could arrive here via a url that doesn't have the operator name in it?

Reply:
yup, anything starting with install will get here. This should handle that case though and return an error indicating an invalid operator name.